### PR TITLE
(PDB-1034) Ezbake source based testing

### PR DESCRIPTION
**Pull request by kbarber**
This change switched PuppetDB over to use the ezbake_utils helpers in ezbake
to install PuppetDB via source. This effectively means the old installation
methodology can be retired.

The beaker-tests-source.sh script is now modified to ensure that leiningen
is now installed (since its required by ezbake helpers in beaker).

Some minor code was removed that is no longer needed, but a major retirement
patch will be forthcoming to remove the greater amount of code no longer
required. Since it was deemed such a patch would make this patch hard to read.

Signed-off-by: Ken Barber <ken@bob.sh>

### CODE REVIEW

**Comment by rbrw** [comment]
@pljenkinsro retest this please

**Comment by kbarber** [comment]
maven transient ^^

File: acceptance/README.md [hunk 1]
```diff
@@ -116,10 +114,6 @@ in your hash; the environment variable names are the same but uppercased
   from source. This is primarily so we can test against alternate versions of
   Puppet, so if the Puppet repo is not specified we fall back to using packages.
 
-* `:puppetdb_repo_puppetdb` (`PUPPETDB_REPO_PUPPETDB`) :
```
Comment (senior) [inline acceptance/README.md]: Do we just use the beaker arg for this functionality now?

Comment (kbarber) [inline acceptance/README.md]: It now does a local tar.gz build, transfers that to the remote host, then unpacks it there etc. it not longer needs to have the repo, as in no longer needs to do a secondary pull of that repo. It uses the local repo for all its needs.

**Pull request merged**